### PR TITLE
Add MultipleNegativesRankingLoss

### PR DESCRIPTION
**Pull request by mcschmitz**
Adds the `MultipleNegativesRankingLoss` as in #664 

The [reference paper](https://arxiv.org/pdf/1705.00652.pdf) assumes one single positive per anchor. However, if there are multiple positive samples for an anchor, this implementation calculates the loss for all combinations of anchors and positives and takes the average by anchor.
This could theoretically be bypassed by using a miner, that only returns the hardest positive.

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Thanks for creating this PR!

After taking a closer look, I think this is equivalent to NTXentLoss with a customized reducer. Here's a script I used to check:

```python
import sys

sys.path.insert(0, "src")
import torch

from pytorch_metric_learning.losses import MultipleNegativesRankingLoss, NTXentLoss
from pytorch_metric_learning.reducers import AvgNonZeroReducer, PerAnchorReducer

l1 = MultipleNegativesRankingLoss(scale=10)
l2 = NTXentLoss(temperature=0.1, reducer=PerAnchorReducer(AvgNonZeroReducer()))

for _ in range(100):
    embeddings = torch.randn(32, 128)
    labels = torch.randint(0, 10, size=(32,))
    loss1 = l1(embeddings, labels)
    loss2 = l2(embeddings, labels)
    print(loss1, loss2)
    if not torch.isclose(loss1, loss2):
        raise ValueError(f"{labels}")
```

Let me know if you get the same results.

**Comment by KevinMusgrave** [comment]
I'm going to close this for now. Let me know if you get different results, and I'll reopen.

**Comment by mcschmitz** [comment]
Sorry for getting back to you just now. I double-checked your script and can confirm it yields the same results as the `MultipleNegativesRankingLoss`. Thanks for clarifying this!
